run-command.c

```diff
@@ -795,6 +796,11 @@ int start_command(struct child_process *cmd)
 		if (cmd->dir && chdir(cmd->dir))
 			child_die(CHILD_ERR_CHDIR);
 
+		if (cmd->close_parent_file_descriptors) {
+			for (fd = 3; fd < 256; ++fd)
```

Comment:
If you want to upstream this, change `++fd` to `fd++`. For some reason, the Git developers prefer this.

About the magic `256`... This sent me down a fun rabbit hole, thank you. The short version is that you should replace the constant by [`getrlimit(RLIMIT_NOFILE, &fd_max_plus_one)`](https://pubs.opengroup.org/onlinepubs/009695399/functions/getrlimit.html).

But wait! Is `getrlimit()` available everywhere? And even more importantly: does Git already use that function? Let's find out... `git grep getrlimit` to the rescue.

Turns out that my short version was wrong: there is a `get_max_fd_limit()` function in `packfile.c`. What you will want to do is to move it to a more sensible place, declare it as a global function in `cache.h`, and then use it also here.

(BTW typically, the `RLIMIT_NOFILE` value seems to be 1024... and beware: `get_max_fd_limit()` returns `1` as a fall-back, if there is no other available means to determine the value.)